src/java/org/apache/cassandra/io/util/PathUtils.java

```diff
@@ -63,14 +63,16 @@
     private static final Set<StandardOpenOption> READ_WRITE_OPTIONS = unmodifiableSet(EnumSet.of(READ, WRITE, CREATE));
     private static final FileAttribute<?>[] NO_ATTRIBUTES = new FileAttribute[0];
 
+    private static final boolean USE_NIX_RECURSIVE_DELETE = CassandraRelevantProperties.USE_NIX_RECURSIVE_DELETE.getBoolean();
```

Comment:
Could this condition be automatic on *nix, instead of explicit via a property flag?

Is this applicable to runtime as well as tests?

Reply:
I wanted to make this a simple fix for testings - I guess changing that in general would require some extended verification

Reply:
fair enough. maybe a comment that we eventually want to hard code it to true? 
(and how we will encourage testing/verification of it?)

Reply:
Looking into other classes I perceived it as a convention, but I'm not opposed to change that

Reply:
Eh, either way is fine. Just curious.